cmd/oras/internal/display/status/track/reader.go

```diff
@@ -34,28 +42,20 @@ type reader struct {
 }
 
 // NewReader returns a new reader with tracked progress.
-func NewReader(r io.Reader, descriptor ocispec.Descriptor, actionPrompt string, donePrompt string, tty *os.File) (*reader, error) {
+func NewReader(r io.Reader, descriptor ocispec.Descriptor, actionPrompt string, donePrompt string, tty *os.File) (Reader, error) {
 	manager, err := progress.NewManager(tty)
 	if err != nil {
 		return nil, err
 	}
-	return managedReader(r, descriptor, manager, actionPrompt, donePrompt)
-}
-
-func managedReader(r io.Reader, descriptor ocispec.Descriptor, manager progress.Manager, actionPrompt string, donePrompt string) (*reader, error) {
-	messenger, err := manager.Add()
-	if err != nil {
-		return nil, err
-	}
-
-	return &reader{
+	tr := reader{
 		base:         r,
 		descriptor:   descriptor,
 		actionPrompt: actionPrompt,
 		donePrompt:   donePrompt,
 		manager:      manager,
-		messenger:    messenger,
-	}, nil
```

Comment:
It's not right to merge `managedReader` into `NewReader`. I would suggest to move it to `target.go`.